enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,349 @@
+---
+title: heterogeneous-architecture-clusters
+authors:
+  - "@Prashanth684"
+reviewers:
+  - "@bparees"
+  - "@derekwaynecarr"
+  - "@jupierce"
+  - "@sdodson"
+  - "@wking"
+  - "@LalatenduMohanty - upgrades"
+  - "@sosiouxme - payload build and release"
+  - "@enxebre - hypershift"
+  - "@dmage - imagestream api changes"
+  - "@kdoberst - console filtering changes"
+  - "@adambkaplan - Openshift Builds"
+  - "@perdasilva - OLM changes"
+  - "@mrunalp - MCO changes"
+approvers:
+  - "@bparees"
+  - "@derekwaynecarr"
+creation-date: 2022-01-17
+last-updated: 2022-01-17
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2240
+  - https://issues.redhat.com/browse/OCPPLAN-4577
+status: implementable
+---
+
+# heterogeneous-architecture-cluster-support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for tech preview and GA
+- [ ] User-facing documentation is created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+## Summary
+
+This enhancement describes the plan to support provisioning and upgrades of heterogeneous architecture Openshift clusters.
+A heterogeneous cluster is a cluster with nodes running on different architectures with a homogeneous control plane and worker node pools
+supporting any combination of architectures (eg: x86 control plane + a mix of ARM/P/Z worker nodes). The node pools within the cluster are
+homogeneous. This document describes the support needed for heterogeneous architecture clusters in phases. There will be three major phases
+with basic support in phase 1 which is targeted for 4.11 and the other phases which address gaps beyond the 4.11 time frame. The targeted
+platforms in the initial phases are Azure and AWS with support for x86 and ARM architectures.
+
+![alt heterogeneous cluster](heterogeneous-cluster.png)
+
+## Motivation
+
+- Users can take advantage of cost savings on architectures like ARM.
+- Users can colocate applications that require different architectures rather than needing multiple clusters and thus avoid cross cluster
+latency and connectivity failures.
+
+### Goals
+
+- Provide a way to install and upgrade clusters with varied architecture node pools with the ability to autoscale heterogeneous machinesets.
```

Comment:
Changes in the machine API to support auto scaling from zero are probably needed to make the autoscaler aware of the upcoming Node arch cc @JoelSpeed @elmiko

Reply:
i'm not sure how the autoscaler handles the architecture of the workload in the scheduler. i'm guessing this is already part of kube's scheduler, so it probably "just works", but i'm guessing we will need to pass the cpu architecture in addition to cpu cores at the least. this will probably require a little investigation into how the autoscaler handles the architecture.

Reply:
makes perfect sense, thanks for the guidance =)

Reply:
will create a machine-api section and put this in. thanks!

Reply:
added the new section for machine api changes. @JoelSpeed @elmiko let me know if there are any changes/addition needed.